compiler/AST/primitive.cpp

```diff
@@ -584,6 +584,7 @@ initPrimitive() {
   prim_def(PRIM_TRY_EXPR, "try-expr", returnInfoFirst);
   prim_def(PRIM_TRYBANG_EXPR, "try!-expr", returnInfoFirst);
   prim_def(PRIM_YIELD, "yield", returnInfoFirst, true);
+  prim_def(PRIM_REDUCE, "reduce", returnInfoVoid, true); // pre-resolve only
```

Comment:
Can you add a whole comment describing what PRIM_REDUCE is and what its arguments are?